ONSdigital/census-rm-print-file-service
# Improve logging, service stability and use enums for constant string sets

# Motivation and Context
Logging level was not being reflected in stackdriver, adding a 'severity' key to the JSON format fixes this. Also ingesting bad messages could cause the message listener to die, it now catches, logs and nacks the message instead.

# What has changed
* Add severity to log JSON
* Catch, log and nack errors processing messages
* Log meaningful errors if messages are malformed
* Use Enums for the sets of constant strings like PackCode and ActionType
* Pull the exceptions out into their own module
* Add some more tests

# How to test?
The happy paths should all function exactly as before. The message listener should now not die but log and error and nack the message if it is missing required metadata keys or has an invalid action type or pack code. If you build and run this version in GCP, you should now see error level logs recorded with the correct severity

# Links
https://trello.com/c/ASmhCdSE/726-rmc-134b1-implement-pq-requested-event-adhoc-fulfillment-requests-for-print-files-in-actions-scheduler-part-2-8
